app/controllers/auth_controller.rb

```diff
@@ -2,7 +2,14 @@ class AuthController < ApplicationController
   with_auth_session_management
 
   def sign_in
-    session[:last_course_id] = params[:last_course_id] if params.key?(:last_course_id)
+    session.delete(:last_course_id)
+    if params.key?(:last_course_id)
+      session[:last_course_id] = params[:last_course_id]
+    elsif params.key?(:course_id)
+      session[:last_course_id] = params[:course_id]
+      session[:forwarding_url] = course_path(params[:course_id])
+    end
```

Comment:
Nice 👍 – if you want, you can pull this out in to a separate PR, so we can get it to production quicker? (I'm assuming it's not strictly relevant/dependent on the rest of the PR).

Reply:
Pulled out into here: https://github.com/chaynHQ/soulmedicine/pull/162

Reply:
Merged into master now.